src/group.ts

```diff
@@ -356,110 +182,19 @@ export class Elt {
-            const tv2 = u.mul(v) //         2. tv2 = u * v
-            tv1 = tv1.mul(tv2) //           3. tv1 = tv1 * tv2
-            let y1 = tv1.power(c1) //       4. y1 = tv1^c1
-            y1 = y1.mul(tv2) //             5. y1 = y1 * tv2
-            const y2 = y1.mul(c2) //        6. y2 = y1 * c2
-            let tv3 = y1.square() //        7. tv3 = y1^2
-            tv3 = tv3.mul(v) //             8. tv3 = tv3 * v
-            const isQR = tv3.equals(u) //   9. isQR = tv3 == u
-            const y = cmov(y2, y1, isQR) // 10. y = CMOV(y2, y1, isQR)
-            return { isQR, root: y } //     11. return (isQR, y)
-        }
-
-        let tv1 = u.square() //         1.  tv1 = u^2
-        tv1 = Z.mul(tv1) //             2.  tv1 = Z * tv1
-        let tv2 = tv1.square() //       3.  tv2 = tv1^2
-        tv2 = tv2.add(tv1) //           4.  tv2 = tv2 + tv1
-        let tv3 = tv2.add(one) //       5.  tv3 = tv2 + 1
-        tv3 = B.mul(tv3) //             6.  tv3 = B * tv3
-        let tv4 = cmov(Z, zero.sub(tv2), !tv2.equals(zero)) // 7.  tv4 = CMOV(Z, -tv2, tv2 != 0)
-        tv4 = A.mul(tv4) //             8.  tv4 = A * tv4
-        tv2 = tv3.square() //           9.  tv2 = tv3^2
-        let tv6 = tv4.square() //       10. tv6 = tv4^2
-        let tv5 = A.mul(tv6) //         11. tv5 = A * tv6
-        tv2 = tv2.add(tv5) //           12. tv2 = tv2 + tv5
-        tv2 = tv2.mul(tv3) //           13. tv2 = tv2 * tv3
-        tv6 = tv6.mul(tv4) //           14. tv6 = tv6 * tv4
-        tv5 = B.mul(tv6) //             15. tv5 = B * tv6
-        tv2 = tv2.add(tv5) //           16. tv2 = tv2 + tv5
-        let x = tv1.mul(tv3) //         17.   x = tv1 * tv3
-        const { isQR, root: y1 } = sqrt_ratio_3mod4(tv2, tv6) // 18. (is_gx1_square, y1) = sqrt_ratio(tv2, tv6)
-        let y = tv1.mul(u) //           19.   y = tv1 * u
-        y = y.mul(y1) //                20.   y = y * y1
-        x = cmov(x, tv3, isQR) //       21.   x = CMOV(x, tv3, is_gx1_square)
-        y = cmov(y, y1, isQR) //        22.   y = CMOV(y, y1, is_gx1_square)
-        const e1 = sgn(u) === sgn(y) // 23.  e1 = sgn0(u) == sgn0(y)
-        y = cmov(zero.sub(y), y, e1) // 24.   y = CMOV(-y, y, e1)
-        const z = tv4 //                25.   x = x / tv4
-        x = x.mul(z) //                 26. return (x, y, z)
-        tv1 = z.square()
-        tv1 = tv1.mul(z)
-        y = y.mul(tv1)
-
-        const point = new sjcl.ecc.pointJac(curve, x, y, z).toAffine()
-        if (!point.isValid()) {
-            throw new Error('point not in curve')
-        }
-        return new Elt(g, point)
-    }
-
     static async hash(g: Group, msg: Uint8Array, dst: Uint8Array): Promise<Elt> {
-        const u = await Elt.hashToField(g, msg, dst, 2)
-        const Q0 = Elt.sswu(g, u[0])
-        const Q1 = Elt.sswu(g, u[1])
-        return Q0.add(Q1)
+        const h2c = getCurve(g.id).h2c
+        const DST = new TextDecoder().decode(dst)
```

Comment:
`DST` must be an array of bytes, not a string. (In the h2c draft we used ascii strings for simplicity).

The change must come from here: https://github.com/paulmillr/noble-curves/blob/main/src/abstract/hash-to-curve.ts#L130